presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/JoinMatcher.java

```diff
@@ -44,6 +48,21 @@ public boolean matches(PlanNode node, Session session, Metadata metadata, Expres
             if (joinNode.getType() != joinType) {
                 return false;
             }
+
+            if (filter.isPresent()) {
```

Comment:
this is somehow complicated `if` blocks

what would you think of something like that:

```
if (filter.isPresent()) {
  if (!join.getFilter().isPresent()) {
    return false;
  } 

  return new ExpresionVeriffier...();
}
```